commit 5148da09dcd3f00913ddfc5d03901c4de56b61e3
Author: Francois Dugast <francois.dugast@intel.com>
Date:   Wed Mar 5 16:06:59 2025 +0100

    drm/xe: Allow fault injection in exec queue IOCTLs
    
    Use fault injection infrastructure to allow specific functions to
    be configured over debugfs for failing during the execution of
    xe_exec_queue_create_ioctl(). xe_exec_queue_destroy_ioctl() and
    xe_exec_queue_get_property_ioctl() are not considered as there is
    no unwinding code to test with fault injection.
    
    This allows more thorough testing from user space by going through
    code paths for error handling and unwinding which cannot be reached
    by simply injecting errors in IOCTL arguments. This can help
    increase code robustness.
    
    The corresponding IGT series is:
    https://patchwork.freedesktop.org/series/144138/
    
    Reviewed-by: Sai Teja Pottumuttu <sai.teja.pottumuttu@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20250305150659.46276-1-francois.dugast@intel.com
    Signed-off-by: Francois Dugast <francois.dugast@intel.com>

diff --git a/drivers/gpu/drm/xe/xe_vm.c b/drivers/gpu/drm/xe/xe_vm.c
index 3dbd3d38008a..338d98533fae 100644
--- a/drivers/gpu/drm/xe/xe_vm.c
+++ b/drivers/gpu/drm/xe/xe_vm.c
@@ -270,6 +270,7 @@ int xe_vm_add_compute_exec_queue(struct xe_vm *vm, struct xe_exec_queue *q)
 
 	return err;
 }
+ALLOW_ERROR_INJECTION(xe_vm_add_compute_exec_queue, ERRNO);
 
 /**
  * xe_vm_remove_compute_exec_queue() - Remove compute exec queue from VM